File: apps/api-nestjs/src/modules/users/dto/update-profile.dto.ts
import { IsString, IsOptional, MinLength, MaxLength, IsBoolean } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';

export class UpdateProfileDto {
  @ApiPropertyOptional({
    description: 'User full name',
    example: 'Jane Smith',
    minLength: 1,
    maxLength: 255
  })
  @IsOptional()
  @IsString()
  @MinLength(1, { message: 'Full name cannot be empty' })
  @MaxLength(255, { message: 'Full name must be at most 255 characters' })
  fullName?: string;

  @ApiPropertyOptional({
    description: 'Unlink local account (username/password). Requires at least one other OAuth identity (Google or Microsoft) to be linked.',
    example: true
  })
  @IsOptional()
  @IsBoolean()
  unlinkLocal?: boolean;
}

File: apps/api-spring/src/main/java/com/todoapp/interfaces/dto/user/UpdateProfileDto.java
package com.todoapp.interfaces.dto.user;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Size;

public record UpdateProfileDto(
    @Email(message = "Invalid email format")
    String email,
    
    @Size(min = 8, message = "Password must be at least 8 characters")
    String password,
    
    @Size(min = 2, max = 100, message = "Full name must be between 2 and 100 characters")
    String fullName
) {}
